util/dvsim/sim_utils.py

```diff
@@ -99,3 +100,90 @@ def vcs_cov_summary_table(buf):
+        if tool == 'xcelium':
+            return xcelium_runtimes(f)
+        elif tool == 'vcs':
+            return vcs_runtimes(f)
+        else:
+            # TODO: support using default dvsim run time
+            raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def vcs_runtimes(buf):
+    """Capture and return two runtimes: job_runtime and simulated_time from VCS simulator.
+
+    Example of VCS job_runtime:    'CPU Time:      1.210 seconds;'
+    Example of VCS simulated_time: '$finish at simulation time 522104678 ps'
+    """
+    job_runtime = 0
+    simulated_time = 0
+
+    for line in buf:
+        # Extract job runtime.
+        if "CPU Time" in line:
+            # Find pattern: CPU Time: "XXXX" seconds;.
+            m = re.search(r"(\d+\.\d+)", line)
+            if m:
+                job_runtime = float(m.group(0))
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+
+        # Extract sim runtime.
+        if "finish at simulation time" in line:
+            # Find pattern: finish at simulation time "XXXX" ps.
+            m = re.search(r"(\d+)", line)
+            if m:
+                simulated_time = "{:.3f}".format(int(m.group(0)) / 1_000_000_000.0)
+            else:
+                raise SyntaxError(f"Cannot find sim runtime in line: " + line)
+
+        if job_runtime and simulated_time:
+            return job_runtime, simulated_time
+    log.warning("Fail to parse VCS runtimes!")
+
+
+def xcelium_runtimes(buf):
+    """Capture and return two runtimes: job_runtime and simulated_time from Xcelium simulator.
+
+    Example of Xcelium job_runtime:
+      'TOOL: xrun(64) 21.09-s006: Exiting on Jul 22, 2022 at 13:38:48 PDT (total: 00:00:02)'
+    Example of Xcelium simulated_time:
+      'UVM_INFO @  51524469 ps: (uvm_report_server.svh:847) [UVM/REPORT/SERVER]'
+    """
+    job_runtime = 0
+    simulated_time = 0
+
+    for line in buf:
+        # Extract job runtime.
+        if "total:" in line and "TOOL" in line:
+            # Find pattern: TOOL .. (total: "XX":"XX":"XX")
+            m = re.search(r"total:.*(\d+):(\d+):(\d+)", line)
+            if m:
+                # Convert time format: HR:MIN:SEC to seconds.
+                val = re.findall(r"\d+", m.group(0))
+                job_runtime = (int(val[0]) * 60 + int(val[1])) * 60 + int(val[2])
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+
+        # Extract sim runtime.
+        if "uvm_report_server" in line:
+            m = re.search(r"(\d+).*ps", line)
+            if m:
+                simulated_time = "{:.3f}".format(int(m.group(1)) / 1_000_000_000.0)
+            else:
+                raise SyntaxError(f"Cannot find sim runtime in line: " + line)
+
+        if job_runtime and simulated_time:
+            return job_runtime, simulated_time
+    log.warning("Fail to parse Xcelium runtimes!")
```

Comment:
This warning will be printed for as many tests as we are running - which could be in the 1000s. So the dvsim stdout will be flooded with 1000s of this warning message.

Reply:
Oh! Did I misunderstand somethng here? I thought this warning will only be printed out if we did not extract the `runtimes` successfully. Because line 188 should return and exit this function if we find the correct value?

Reply:
Right - imagine we had a non-UVM testbench (`prim_prince` for example), which has 1 test, but we ran 100s of seeds. If we run with Xcelium, it will print this warning 100s of times.

Reply:
Oh! Ouch I see your point! Totally agree! Should only be printed out once per test max! Let me try fixing that